PORTAL/jobs/requests.py

```diff
@@ -389,18 +416,18 @@ def set_status(self, status):
         if status in self.FINISHED:
             self.close()
 
-    def close(self):
+    def close(self) -> None:
         if self.history[-1][0] is self.CLOSED:
             # XXX Fail?
             return
         self.history.append(
             (self.CLOSED, datetime.datetime.now(datetime.timezone.utc)),
         )
 
-    def upload(self, req):
+    def upload(self, req: str) -> None:
         raise NotImplementedError
 
```

Comment:
I think I went in a different direction with uploading results.  We could probably drop this method.  (It doesn't have to happen in this PR though.)